coffeecard/CoffeeCard.Library/Services/v2/AccountService.cs

```diff
@@ -210,6 +210,38 @@ public async Task UpdateUserGroup(UserGroup userGroup, int userId)
             await _context.SaveChangesAsync();
         }
 
+        public async Task<List<User>> SearchUsers(String search, int pageNum, int pageLength)
+        {
+            List<User> users = new List<User>();
+
+            int skip = pageNum * pageLength;
+
+            if (int.TryParse(search, out int id))
+            {
+                var user = await _context.Users.FirstOrDefaultAsync(u => u.Id == id);
+                if (user != null)
+                {
+                    users.Add(user);
+                    return users;
+                }
+            }
+
+            users = await _context.Users
+                .Where(u => u.Email.ToLower().StartsWith(search.ToLower()) || u.Name.ToLower().StartsWith(search.ToLower()))
+                .OrderBy(u => u.Id)
+                .Skip(skip).Take(pageLength)
+                .ToListAsync();
```

Comment:
Greatly reduce code complexity and eliminate unintended edge cases (you may think they're stupid, but they're important) with this change:

```suggestion
            var users = await _context.Users
                .Where(u => u.Email.ToLower().StartsWith(search.ToLower()) || u.Name.ToLower().StartsWith(search.ToLower()) || u.Id.ToString().StartsWith(search))
                .OrderBy(u => u.Id)
                .Skip(skip).Take(pageLength)
                .ToListAsync();
```

alternatively, to force exact matches on the id, do this comparison instead:
```
u.Id.ToString() == search
```

Reply:
totally agree, I'll try to find some time before Christmas to apply the changes suggested by both of you :)

Reply:
Also I'm not sure whether we really need pagination. I don't think above suggestions will be that expensive, thus I'm not sure how much we need pagination. :)

Reply:
Thirdly there is currently no Index on name. We should make sure to add an index to the User column to make the query less expensive

Reply:
I was not part of the discussion but would imagine that using Contains over Startswith would give a better experience.

One note on using the Contains method is that it is case sensitive but if you use a SQL `like` method which is case insensitive. To use the like method, you should use the https://learn.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.dbfunctionsextensions.like?view=efcore-8.0

Reply:
I don't think we should remove pagination. Pagination is for the frontend - we discussed a solution where the user sees a list of accounts (where they can change user roles) and then a search bar which allows filtering. This means we can avoid having to send a lot of data from the API as we only send `pageLength` accounts.

As for the input length - I don't think we should limit it. We want to show the first `pageLength` accounts on the initial page. I.e., we want to make a "search" for the empty input.

Reply:
@A-Guldborg Thanks for providing insight into the previous discussion and use case for this new endpoint. I agree that this pagination is relevant

Reply:
@jonasanker An issue with conditionally determining search column is: what if the  user's name is "1" or "1@" (both legal names as of writing)? You wouldn't be able to find user via their name with the queries "1" (would look in the Ids instead) or "1@" (would look in emails instead). Searching in an additional two columns sounds like a good cost of covering these edge cases, considering the expected use frequency of this endpoint